File: Java/SlidingWindowAndTwoPointers/1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.java
class Solution {
    public int numberOfSubstrings(String s) {
        int n = s.length();
        int lastseen[] = {-1, -1, -1};
        int maxcnt=0;
        for(int i=0;i<n;i++){
            lastseen[s.charAt(i)-'a'] = i;
            int minIndex = Math.min(lastseen[0], Math.min(lastseen[1], lastseen[2]));
            maxcnt += (minIndex + 1);
        }
        return maxcnt;
    }
}

File: Python/SlidingWindowAndTwoPointers/Medium/1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        lastseen = [-1,-1,-1]
        n = len(s)
        maxcnt=0
        for i in range(n):
            lastseen[ord(s[i]) - ord('a')]  = i
            if lastseen[0] != -1 and lastseen[1] != -1 and lastseen[1] != -1:
                maxcnt = maxcnt + 1+ min(lastseen[0],lastseen[1],lastseen[2])
        
        return maxcnt